### PR TITLE
feat(context-dialog): Added footer to context dialog.

### LINKED ISSUE #322: Context dialog improvements
# Feature Request

## Summary

A scrollable content section for the context dialog.

## Feature Description

Problem: If there is too much content in a context dialog the buttons on the bottom are not visible.

Solution: The dialog should have a max-height of about 90vh. A sticky footer section ensures, that the buttons are always visible. The content of the dialog must then be scrollable.
- [ ]  Extend api with a footer that is sticky
- [ ]  Add closing button on the top right corner besides the title. 
- [ ] Make the content section scrollable
- [ ] Provide a mechanism for the consumer to set the OverlayConfig used by the context dialog, keep in mind that this config includes things like width, height, ...

## Attachments

image

context-dialog-scrollable

### PR DESCRIPTION
**Pull request by NinaKammerlander**
### <strong>Pull Request</strong>

Added a sticky footer for the context dialog, changed the content to be scrollable and reimplemented the closing button. Additionally added the possibility to provide an injection token, to overwrite certain configuration of the context dialog.

closes #322 
<hr>

#### Type of PR

 Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomheller** [comment]
Pull request looks good to me. Good job on dealing with all these dialog-edge-cases 👍

**Comment by ffriedl89** [comment]
Merging since the flaky e2e test is fixed in #584

**Comment by rowa-audil** [comment]
@NinaKammerlander Resolve conflicting files, please :)

File: apps/components-e2e/src/components/context-dialog/context-dialog.e2e.ts [hunk 1]
```diff
@@ -23,32 +23,40 @@ import {
 
 import { getActiveElementText } from '../overlay/overlay.po';
 
-fixture('Context Dialog').page('http://localhost:4200/context-dialog');
+fixture
+  .only('Context Dialog')
```
Comment (tomheller) [inline apps/components-e2e/src/components/context-dialog/context-dialog.e2e.ts]:
```suggestion
  .('Context Dialog')
```

**Comment by tomheller** [comment]
Manually cherry picked to 5.x
